### PR TITLE
Mount option add entry to fstab

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Creates a mount point entry in `/etc/fstab` when needed.
- Rule `mount_option_dev_shm_noexec` can remediate, by creating an entry in `/etc/fstab`.
- Disable creation of `fstab` for removable partitions and remote filesystems, we don't have enough information for that.

#### Rationale:

- By default `/dev/shm/` is mounted by systemd with `nosuid` and `nodev` options.

#### Issues

After a scan and remediation, `/etc/fstab` contains only `noexec` option, because rules `mount_option_dev_shm_nosuid` and `mount_option_dev_shm_nodev` pass, and therefore, remediation script is not run.
~There is some concurrence/caching occurring, either on scanner side or kernel side.~

~The `noexec` rule is the first to run, and its remediation should cause the other two rules to detect the mount as *without* `nosuid` and `nodev` options. But the scanner evaluates the old table and passes.~

Edit: There is no caching issue. The`nosuid` and `nodev` rules are not remediated because they **passed** during scan, so no remediation is performed.

**Again** we have issues with rule inter dependency.
If the scanner did "scan rule/fix rule" instead of doing "scan all rules/fix all rules", this would be merged already. :D

### CODE REVIEW

**Comment by yuumasato** [comment]
@openscap-ci  test this please

File: ol7/templates/csv/mount_options.csv [hunk 1]
```diff
@@ -9,6 +9,6 @@
 
 # /dev/shm is created by systemd and is not available at install time
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply ol7/templates/csv/mount_options.csv]: Comment documentation updated.

File: shared/bash_remediation_functions/include_mount_options_functions.sh [hunk 2]
```diff
@@ -4,23 +4,30 @@ function include_mount_options_functions {
...
+# $4: mount type of mount point (used when adding new entry in fstab)
 function ensure_mount_option_in_fstab {
-	local _mount_point="$1" _new_opt="$2" _mount_point_match_regexp="" _previous_mount_opts=""
+	local _mount_point="$1" _new_opt="$2" _filesystem=$3 _type=$4
+	local _mount_point_match_regexp="" _previous_mount_opts=""
 	_mount_point_match_regexp="$(get_mount_point_regexp "$_mount_point")"
 
-	if [ $(grep "$_mount_point_match_regexp" /etc/fstab | grep -c "$_new_opt" ) -eq 0 ]; then
+	if [ $(grep -c "$_mount_point_match_regexp" /etc/fstab ) -eq 0 ]; then
+        echo "${_filesystem} ${_mount_point} ${_type} defaults,${_new_opt} 0 0" >> /etc/fstab
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply shared/bash_remediation_functions/include_mount_options_functions.sh]: Fixed.

**Comment by yuumasato** [comment]
Given that the issue is not an environment or bug in scanner, but a limitation of it (more likely a limitation of standard specification), I consider it good to (review and) merge.

**Comment by jan-cerny** [comment]
@yuumasato Thanks. What is your opinion on merging the 3 rules in a single rule which would run the remediation at once and therefore solve the problem?

**Comment by yuumasato** [comment]
> @yuumasato Thanks. What is your opinion on merging the 3 rules in a single rule which would run the remediation at once and therefore solve the problem?

I don't think that is the way to go, each rule is different setting and  has a unique CCE assigned to it.

**Comment by jan-cerny** [comment]
@yuumasato Great job! Thank you.

**Pull request merged**